Backend/DataAccessLayer/Repositories/UserManagementRepository.cs

```diff
@@ -11,85 +11,56 @@ namespace DataAccessLayer.Repositories
 {
     public class UserManagementRepository
     {
-        public int CreateNewUser(User user)
+        public User CreateNewUser(User user, DatabaseContext _db)
         {
-            using (var _db = new DatabaseContext())
-            {
-                try
-                {
-                    _db.Users.Add(user);
-                    return _db.SaveChanges();
-                }
-                catch (Exception)
-                {
-                    return 0;
-                }
-            }
+            _db.Entry(user).State = EntityState.Added;
+            return user;
         }
 
-        public int DeleteUser(Guid Id)
+        public User DeleteUser(Guid Id, DatabaseContext _db)
         {
-            using (var _db = new DatabaseContext())
-            {
-                var user = _db.Users
-                    .Where(c => c.Id == Id)
-                    .FirstOrDefault<User>();
-                if (user == null)
-                    return 0;
-                _db.Entry(user).State = EntityState.Deleted;
-                return _db.SaveChanges();
-            }
+            var user = _db.Users
+                .Where(c => c.Id == Id)
+                .FirstOrDefault<User>();
+            if (user == null)
+                return null;
+            _db.Entry(user).State = EntityState.Deleted;
+            return user;
         }
 
-        public User GetUser(string email)
+        public User GetUser(string email, DatabaseContext _db)
         {
-            using (var _db = new DatabaseContext())
-            {
-                var user = _db.Users
-                    .Where(c => c.Email == email)
-                    .FirstOrDefault<User>();
-                return user;
-            }
+            var user = _db.Users
+                .Where(c => c.Email == email)
+                .FirstOrDefault<User>();

```

Comment:
Hmm, firstordefault. What does this return if not found? An empty (default) user?

Reply:
default is null